src/cmap/connection.ts

```diff
@@ -594,11 +594,33 @@ export class CryptoConnection extends Connection {
       return;
     }
 
+    // Save sort or indexKeys based on the command being run
```

Comment:
This doesn't feel like the correct solution.  If I understand the issue correctly, when libmongocrypt sends back an encrypted command as a serialized bson object, we deserialize it into a JS object to send to the server.  Why wouldn't we fix the problem by deserializing into a map instead of an object in mongodb-client-encryption instead of working around it in the driver?

Reply:
Would the code not look the same just in `mongodb-client-encryption`? The layer at which BSON is deserialized is still inside the BSON library so performing the logic seen here would be nearly identical, just within the `encrypt` method. Since it's something we would have to test in this repo anyway, (at least integration test) feels like the right spot is here.

Reply:
I was imagining we'd build a feature into bson to serialize directly into maps instead of objects (if it doesn't exist), and then configure bson to use that for everything in FLE.  With eventual on-demand-bson we could combine this approach with that, and selectively deserialize directly into maps but handle it in bson with a configuration.